tests/tests_settings.yml

```
@@ -0,0 +1,115 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Test bootloader_settings variable
+  hosts: all
+  gather_facts: false
+  tags:
+    - tests::reboot
+  vars:
+    bootloader_reboot_ok: true
+  tasks:
+    - name: Replace configuration with settings
+      vars:
+        bootloader_settings:
+          - name: console
+            value: tty0
+          - name: print-fatal-signals
+            value: 1
+          - name: no_timer_check
+            state: present
+          - name: quiet
+          - name: debug
+          - previous: replaced
+        bootloader_timeout: 6
+      include_role:
+        name: linux-system-roles.bootloader
+
+    - name: Flush handlers
+      meta: flush_handlers
+
+    - name: Ensure bootloader_reboot_required is set to false
+      assert:
+        that: not bootloader_reboot_required
+
+    - name: Verify settings
+      shell: >-
+        set -euo pipefail;
+        grubby --info=DEFAULT;
+        grubby --info=DEFAULT |
+        grep -P
+        'args=".*console=tty0 print-fatal-signals=1 no_timer_check quiet
+        debug( |)"$'
+      changed_when: false
+
+    - name: Verify boot loader timeout configuration
+      command: cat /etc/default/grub
+      failed_when: >-
+        __bootloader_default_grub_content | regex_search('^GRUB_TIMEOUT=6')
```

```suggestion
        __bootloader_default_grub_content | regex_search('^GRUB_TIMEOUT=6') | length > 0
```
I think ansible 2.9 cannot evaluate the `list` or `enumeration` returned by `regex_search` in a boolean context